core/store/models/log_events.go

```diff
@@ -85,21 +85,33 @@ func TopicFiltersForRunLog(logTopics []common.Hash, jobID *ID) [][]common.Hash {
 }
 
 // FilterQueryFactory returns the ethereum FilterQuery for this initiator.
-func FilterQueryFactory(i Initiator, from *big.Int) (ethereum.FilterQuery, error) {
+func FilterQueryFactory(i Initiator, nextHead *big.Int) (ethereum.FilterQuery, error) {
 	q := ethereum.FilterQuery{
-		FromBlock: from,
+		FromBlock: nextHead,
 		Addresses: utils.WithoutZeroAddresses([]common.Address{i.Address}),
 	}
 
 	switch i.Type {
 	case InitiatorEthLog:
-		if from == nil {
+		if nextHead == nil {
 			q.FromBlock = i.InitiatorParams.FromBlock.ToInt()
-		} else if from != nil && i.InitiatorParams.FromBlock != nil {
-			q.FromBlock = utils.MaxBigs(from, i.InitiatorParams.FromBlock.ToInt())
+		} else if nextHead != nil && i.InitiatorParams.FromBlock != nil {
+			q.FromBlock = utils.MaxBigs(nextHead, i.InitiatorParams.FromBlock.ToInt())
 		}
 		q.ToBlock = i.InitiatorParams.ToBlock.ToInt()
-		q.Topics = i.Topics
+
+		if q.FromBlock != nil && q.ToBlock != nil && q.FromBlock.Cmp(q.ToBlock) >= 0 {
+			return q, fmt.Errorf("cannot generate a FilterQuery with fromBlock >= toBlock")
+		}
+
+		// It's necessary to manually copy the contents of .Topics rather than applying a simple
```

Comment:
Go culture advocates minimizing comments in favor of expressing the content in the source code itself :)

Reply:
I'm fine with being overruled on this, but I'm also not afraid to disagree with Go culture on a variety of points, comments being one of the more major ones. 🤷🏻‍♂️

It's true that the tests would surface a regression, but the errors they spit out make it very non-obvious what's causing the failure.  Given how long it took Steve and I to work through this bug, and how confused we still are about the solution, it felt appropriate to hang a "here be dragons" sign above it, at least for the time being.

Reply:
It's probably worth filing a bug on this, since it's a mismatch between the compiler's perception of the type, and the running program's.

A slightly terser version which works in my hands is 

```
		q.Topics = make([][]common.Hash, len(i.Topics))
		copy(q.Topics, i.Topics) // Simply coercing i.Topics to the underlying type confuses reflect.DeepEqual
```